### DISCUSSION TITLE
Hostname in Keycloack

### DISCUSSION THREAD

**Discussion by benjaminbascary97** [ragapp/ragapp#146] [General]
Right now i'm testing the `multiple-deployments` feature in an EC2 instance. I'm thinking if it would be a good idea to inject a variable to replace the localhost in the multiple-deployments/docker-compose.yml file. As it is, traeffik (or keycloak?) has issues redirecting in a non local environments and is not possible to utilize it in other environments. The line would be:

`- traefik.http.middlewares.ragapp-keycloakopenid.plugin.keycloakopenid.KeycloakURL=http://localhost/auth`

In the docker-compose file.

Any thoughts?

**Comment by marcusschiesser** [comment]
Yes, it's actually also here:

```
keycloak:
    image: quay.io/keycloak/keycloak:25.0.2
    command: start-dev --import-realm --hostname=http://localhost/auth
```

so

```
keycloak:
    image: quay.io/keycloak/keycloak:25.0.2
    command: start-dev --import-realm --hostname=${KEYCLOAK_HOSTNAME:-http://localhost/auth}
```

might be good replacement

**Reply by benjaminbascary97** [reply]
replaced both with "http://<ec2_instance_Public_IPv4_address>/auth" and the Keycloak dashboard works as expected, previously I had to disable SSL inside the keycloak image stated in this guide:

https://www.adaltas.com/en/2023/03/14/ec2-deploy-keycloak/

As I said, the Keycloak dashboard works, but when I try to reach for the manager app at /manager, if fails with an "invalid redirect URL" error at this URL:

`
http://3.12.6.253/auth/realms/ragapp/protocol/openid-connect/auth?client_id=ragapp&redirect_uri=http%3A%2F%2F3.12.6.253%2Fmanager&response_type=code&state=eyJyZWRpcmVjdF91cmwiOiJodHRwOi8vMy4xMi42LjI1My9tYW5hZ2VyIn0%3D
`

Image:
![imagen](https://github.com/user-attachments/assets/e7d4ef47-c400-47c6-93c7-588f0944a621)

Logs:

![Screenshot 2024-08-23 145120](https://github.com/user-attachments/assets/80c5e1bc-63c5-4e94-a657-dc8392e71cbe)

Do you have any ideas in mind? Im lost here

**Comment by marcusschiesser** [comment]
Will add SSL in https://github.com/ragapp/ragapp/issues/158

**Reply by benjaminbascary97** [reply]
managed to mount a reverse proxy with nginx mapping to 80 to 443 with a self signed certificate but OpenID does not like that either

**Reply by marcusschiesser** [reply]
SSL/TLS is now added in https://github.com/ragapp/ragapp/issues/158 - but you need to use signed certificates - I'm unsure how to make keyclock work with self-signed certificates. Happy to merge a PR if someone finds out

**Reply by benjaminbascary** [reply]
Opened a PR https://github.com/ragapp/ragapp/pull/170. I have written some docs based on my own experience trying to deploy this on EC2 instance

**Comment by Rauljauregi** [comment]
I recently faced similar challenges while setting up my application, and after a lot of trial and error, I found a solution that worked for me:

Instead of making changes to Traefik or Keycloak, I focused on the domain configuration. By routing my domain (or subdomain in my case) through Cloudflare, I set the DNS to Proxy Off and enabled SSL in Flexible mode. This setup allowed me to access the application without any issues.

Hopefully, this approach can help others who are struggling with similar problems. If anyone else has more tips or experiences to share, I'd love to hear them!
